// esly-qurity-sdk/dist/src/EaseSDKCore.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const socket_io_client_1 = __importDefault(require("socket.io-client"));
class EaseSDKCore {
    constructor({ apiKey, email, baseUrl, wsUrl, debug = false }) {
        this.socket = null;
        this.streamId = null;
        this.connected = false;
        if (!apiKey || !email || !baseUrl)
            throw new Error("Error please contact admin");
        this.apiKey = apiKey;
        this.email = email;
        this.baseUrl = baseUrl.replace(/\/$/, "");
        this.wsUrl = wsUrl !== null && wsUrl !== void 0 ? wsUrl : this.baseUrl.replace(/^http/, "ws");
        this.debug = debug;
    }
    async reportIncident({ audio }) {
        const formData = new FormData();
        if (audio)
            formData.append("audio", audio, "recording.wav");
        const res = await fetch(`${this.baseUrl}/api/sdk/incidents`, {
            method: "POST",
            headers: {
                Authorization: `Bearer ${this.apiKey}:${this.email}`,
            },
            body: formData,
        });
        if (!res.ok)
            throw new Error("Failed to report incident");
        return res.json();
    }
    connectStream(streamId) {
        return new Promise((resolve, reject) => {
            try {
                this.streamId = streamId;
                this.socket = (0, socket_io_client_1.default)(`${this.wsUrl}/sdk`, {
                    transports: ["websocket"],
                    query: { apiKey: this.apiKey, streamId },
                });
                this.socket.on("connect", () => {
                    this.connected = true;
                    resolve(true);
                });
                this.socket.on("disconnect", () => {
                    this.connected = false;
                });
                this.socket.on("error", reject);
            }
            catch (err) {
                reject(err);
            }
        });
    }
    sendAudioChunk({ audioBase64, seq, final = false, }) {
        if (!this.connected || !this.socket) {
            throw new Error("Not connected");
        }
        this.socket.emit("audio_chunk", { audioBase64, seq, final });
    }
    endStream() {
        if (this.socket) {
            this.socket.disconnect();
            this.socket = null;
            this.connected = false;
        }
    }
}
exports.default = EaseSDKCore;
