// esly-qurity-sdk/dist/client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EaseClient = void 0;
const utils_1 = require("./utils");
const websocket_1 = require("./websocket");
class EaseClient {
    constructor(opts) {
        if (!opts?.baseUrl || !opts?.apiKey) {
            throw new Error("Error please contact admin");
        }
        this.baseUrl = opts.baseUrl.replace(/\/$/, "");
        this.apiKey = opts.apiKey;
        this.timeoutMs = opts.timeoutMs ?? 15000;
        this.wsUrl = opts.wsUrl;
    }
    // REST API helper
    async fetchJson(path, init) {
        const controller = new AbortController();
        const id = setTimeout(() => controller.abort(), this.timeoutMs);
        try {
            const res = await (0, utils_1.fetchFn)(`${this.baseUrl}${path}`, {
                ...init,
                headers: {
                    ...(init.headers ?? {}),
                    "x-api-key": this.apiKey,
                    "content-type": "application/json",
                },
                signal: controller.signal,
            });
            clearTimeout(id);
            if (!res.ok) {
                const txt = await res.text();
                throw new Error(`Error please contact admin`);
            }
            return (await res.json());
        }
        finally {
            clearTimeout(id);
        }
    }
    // Report single incident
    async reportIncident(payload) {
        const res = await this.fetchJson("/api/sdk/incidents", {
            method: "POST",
            body: JSON.stringify({
                ...payload,
                timestamp: payload.timestamp ?? new Date().toISOString(),
            }),
        });
        return res;
    }
    // Start live audio stream
    async startAudioStream() {
        if (this.ws)
            return;
        const wsUrl = this.wsUrl ?? this.baseUrl.replace(/^http/, "ws");
        const connectUrl = `${wsUrl}/sdk?apiKey=${encodeURIComponent(this.apiKey)}`;
        this.ws = new websocket_1.EaseWebSocket(connectUrl);
        this.ws.connect();
    }
    // Send a chunk of audio
    sendAudioChunk(chunk) {
        if (!this.ws)
            throw new Error("Error please contact admin");
        this.ws.sendChunk(chunk);
    }
    // Stop the stream
    stopAudioStream() {
        this.ws?.close();
        this.ws = undefined;
    }
    // Check quota
    async getQuota() {
        return await this.fetchJson("/api/sdk/quota", { method: "GET" });
    }
}
exports.EaseClient = EaseClient;
