// esly-qurity-sdk/dist/src/client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EaseClient = void 0;
const utils_1 = require("./utils");
const websocket_1 = require("./websocket");
class EaseClient {
    constructor(opts) {
        var _a;
        if (!(opts === null || opts === void 0 ? void 0 : opts.baseUrl) || !(opts === null || opts === void 0 ? void 0 : opts.apiKey) || !(opts === null || opts === void 0 ? void 0 : opts.email)) {
            throw new Error("Error please contact admin");
        }
        this.baseUrl = opts.baseUrl.replace(/\/$/, "");
        this.apiKey = opts.apiKey;
        this.email = opts.email;
        this.timeoutMs = (_a = opts.timeoutMs) !== null && _a !== void 0 ? _a : 15000;
        this.wsUrl = opts.wsUrl;
    }
    // REST API helper
    async fetchJson(path, init) {
        var _a;
        const controller = new AbortController();
        const id = setTimeout(() => controller.abort(), this.timeoutMs);
        try {
            const res = await (0, utils_1.fetchFn)(`${this.baseUrl}${path}`, {
                ...init,
                headers: {
                    ...((_a = init.headers) !== null && _a !== void 0 ? _a : {}),
                    "x-api-key": this.apiKey,
                    "content-type": "application/json",
                },
                signal: controller.signal,
            });
            clearTimeout(id);
            if (!res.ok) {
                const txt = await res.text();
                throw new Error(`Error please contact admin`);
            }
            return (await res.json());
        }
        finally {
            clearTimeout(id);
        }
    }
    // Report single incident
    async reportIncident({ audio }) {
        const formData = new FormData();
        if (audio)
            formData.append("audio", audio, "recording.wav");
        const res = await fetch(`${this.baseUrl}/api/sdk/incidents`, {
            method: "POST",
            headers: {
                Authorization: `Bearer ${this.apiKey}:${this.email}`, // leave out Content-Type
            },
            body: formData,
        });
        if (!res.ok)
            throw new Error("Failed to report incident");
        return res.json();
    }
    // Start live audio stream
    async startAudioStream() {
        var _a;
        if (this.ws)
            return;
        const wsUrl = (_a = this.wsUrl) !== null && _a !== void 0 ? _a : this.baseUrl.replace(/^http/, "ws");
        const connectUrl = `${wsUrl}/sdk?apiKey=${encodeURIComponent(this.apiKey)}`;
        this.ws = new websocket_1.EaseWebSocket(connectUrl);
        this.ws.connect();
    }
    // Send a chunk of audio
    sendAudioChunk(chunk) {
        if (!this.ws)
            throw new Error("Error please contact admin");
        this.ws.sendChunk(chunk);
    }
    // Stop the stream
    stopAudioStream() {
        var _a;
        (_a = this.ws) === null || _a === void 0 ? void 0 : _a.close();
        this.ws = undefined;
    }
    // Check quota
    async getQuota() {
        return await this.fetchJson("/api/sdk/quota", { method: "GET" });
    }
}
exports.EaseClient = EaseClient;
