// esly-qurity-sdk/dist/src/websocket.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EaseWebSocket = void 0;
const utils_1 = require("./utils");
class EaseWebSocket {
    constructor(wsUrl) {
        this.wsConnected = false;
        this.wsUrl = wsUrl;
    }
    connect(onOpen, onError, onClose) {
        if (this.ws && this.wsConnected)
            return;
        this.ws = new utils_1.WebSocketImpl(this.wsUrl);
        this.ws.onopen = () => {
            this.wsConnected = true;
            onOpen === null || onOpen === void 0 ? void 0 : onOpen();
        };
        this.ws.onerror = (err) => onError === null || onError === void 0 ? void 0 : onError(err);
        this.ws.onclose = () => {
            this.wsConnected = false;
            onClose === null || onClose === void 0 ? void 0 : onClose();
        };
    }
    sendChunk(chunk) {
        if (!this.ws || !this.wsConnected) {
            throw new Error("Error please contact admin");
        }
        this.ws.send(JSON.stringify({ type: "audio_chunk", data: chunk }));
    }
    close() {
        if (this.ws)
            this.ws.close();
        this.ws = undefined;
        this.wsConnected = false;
    }
}
exports.EaseWebSocket = EaseWebSocket;
